### ISSUE TITLE
Review of Codebase

### ISSUE THREAD

**Issue by Ethan-DeBandi99** [Bears-R-Us/arkouda#1957]
In the interest of bringing down compile times and keeping our codebase manageable, we need to review our Chapel code and determine if there is any code that is no longer being used. 

- Is there any code that is never being used?
- Is there any code that is only used in specific situations and could be made optional?

**Comment by Ethan-DeBandi99** [comment]
Wanted to start some conversation here. I do not think that there is anything we can truly remove. However, I think there are some things that would be best if move to Chapel, but want to see everyones thoughts. 

- Comm Aggregation. I believe after talking with @pierce314159 that the majority of features we use should be available in the Chapel module. @ronawho, @bmcdonald3 - Can you confirm this or is there a reason to maintain the current configuration?
- Codec support. I know the fastest way to get this to customer was to include it in Arkouda. Just comparing to other languages, I would thing this should be handled within Chapel itself. Am I missing something that would not allow this to happen in the future? 

@pierce314159 and I discussed some things with File I/O. However, I have investigated this a bit. I think the only spot we would currently gain anything would be with Parquet. However, I think the complexity and the minimal build time of the C++ support code make this a none issue. Again, CSV we would not gain much. Maybe some nice to haves but nothing that would truly impact our compile times. From my vantage point, IO is going to be a non-starter and should be maintained as is.

I have not spent too much time thinking about things that could be made optional. Again, just looking to start the conversation here and get feedback from others.

**Comment by Ethan-DeBandi99** [comment]
True, that conversation was before bigints were being added. From my perspective (at least for the time being) there is not going to be much we can pull out of Arkouda. There is a possibility of making things optional, but I am hesitant to do so with some of the install issues we have had recently. Does anyone disagree with closing this issue as I do not believe we currently are going to be able to alter much.

**Comment by ronawho** [comment]
The things that come to mind for me are:
 - Can we remove the server side dataframe display code (optimizations that have gone in since that code have made the server and client version perform nearly identically in nightly testing -- https://chapel-lang.org/perf/arkouda/16-node-cs-hdr/?startdate=2022/09/03&enddate=2023/01/27&graphs=dataframedisplayperformance). If we can't remove it completely, would it make sense to only bring it in for optimized builds?
 - There's support for two different sort algs. The arkouda radix sort (LSD) and chapel's two array sort (MSD). So far as I understand this was added to evaluate performance of Chapel's two array sort, but I don't believe it's ahead for any of the benchmarks -- https://chapel-lang.org/perf/arkouda/16-node-cs-hdr/?startdate=2022/05/01&enddate=2023/01/27&suite=sortcases. Could this be removed, or are users relying on it?
 - Could we turn off other arkouda optimizations (coargsort that merges arrays, and things like that) if we're doing a non-optimized build?

I don't know that I like the idea of changing what code/algs get used based on optimization levels since that could cause debugging issues, but maybe we could tie it to a different flag for when you're debugging unrelated things.

**Comment by Ethan-DeBandi99** [comment]
The server-side dataframe code is required. That was added based on a user request to be able to more efficiently view DataFrames. At least for me, I am often testing with DataFrames and would not want to only have that available during optimized builds. I don't think we are going to be able to do much there.

I can look into the sort and check if the users need it.

I agree, I am not a fan of altering things based on optimization level. From my viewpoint, I think with the improvements I am seeing from the flag alteration, I am ok with holding off on any major code changes on this front. With that said, I do believe we should run down the sort things @ronawho mentioned.

**Comment by ronawho** [comment]
> The server-side dataframe code is required. That was added based on a user request to be able to more efficiently view DataFrames. At least for me, I am often testing with DataFrames and would not want to only have that available during optimized builds. I don't think we are going to be able to do much there.

I'm saying I don't think there's a major performance difference between the server and client versions anymore and at least in the nightly benchmark they are nearly as efficient.

**Comment by Ethan-DeBandi99** [comment]
Our users were reporting issues with the client side only that have not been reported since adding. I can check back in with them.
